Sources/Target/POSIX/ELFProcess.cpp

```diff
@@ -374,8 +374,9 @@ ErrorCode ELFProcess::updateInfo() {
   // we also validate that this ELF file is really for the current
   // architecture.
   //
-  if (!ELFSupport::MachineTypeToCPUType(machineType, is64Bit, _info.cpuType,
-                                        _info.cpuSubType))
+  ELFSupport::MachineTypeToCPUType(machineType, is64Bit, _info.cpuType,
+                                   _info.cpuSubType);
+  if (_info.cpuType == kCPUTypeAny)
```

Comment:
Is it a problem to still populate _info's fields if ELFSupport::MachineTypeToCPUType fails?

Reply:
I think what you did here is good. Keeping the early return is probably what we want.